Shippo/Batch.cs

```diff
@@ -38,6 +38,132 @@ public class Batch : ShippoId {
         [JsonProperty (PropertyName = "object_results")]
         public ObjectResults ObjectResults;
 
+        public class Address
+        {
+            [JsonProperty (PropertyName = "object_purpose")]
+            public string ObjectPurpose;
+
+            [JsonProperty (PropertyName = "name")]
+            public string Name;
+
+            [JsonProperty (PropertyName = "company")]
+            public string Company;
+
+            [JsonProperty (PropertyName = "street1")]
+            public string Street1;
+
+            [JsonProperty (PropertyName = "street2")]
+            public string Street2;
+
+            [JsonProperty (PropertyName = "city")]
+            public string City;
+
+            [JsonProperty (PropertyName = "state")]
+            public string State;
+
+            [JsonProperty (PropertyName = "zip")]
+            public string Zip;
+
+            [JsonProperty (PropertyName = "country")]
+            public string Country;
+
+            [JsonProperty (PropertyName = "phone")]
+            public string Phone;
+
+            [JsonProperty (PropertyName = "email")]
+            public string Email;
+
+            public static Address createForPurchase (String purpose, String name, String street1, String street2, String city,
```

Comment:
http://stackoverflow.com/a/10439479 provides a way to ignore a property if it is null. I think you should be able to use existing class with this.